metabase/metabase
# Migrate suggestions to MLv2

Closes https://github.com/metabase/metabase/issues/36233

### Description

Rewrite of `suggest.ts` to MLv2.

Covered areas:

- custom column expression
 
<img width="550" alt="image" src="https://github.com/metabase/metabase/assets/125459446/dace5915-a651-469a-974d-c4f79b8a32bc">

- aggregation custom expression
<img width="581" alt="image" src="https://github.com/metabase/metabase/assets/125459446/9b832d88-7866-4fd7-aa4a-cdc7dccd859e">

Admin: segments and metrics are broken intentionally, a fallback can be used instead of real query:

`ExpressionEditorTextfield.tsx` -> `handleCursorChange`
 
```
query: query ?? legacyQuery.question()._getMLv2Query(),
stageIndex: stageIndex ?? -1,
```

### How to verify

- go to notebook editor
- add custom column
- add aggregation custom expression

Both should work.

All tests passed with a fallback

### Checklist
